src/main/java/seedu/lifeasier/model/tasks/Event.java

```diff
@@ -73,4 +75,21 @@ public void moveAndUpdateRecurrences(LocalDate day) {
             end = end.plusDays(7);
         }
     }
+
+    public boolean isDuplicate(String description, LocalDateTime start, LocalDateTime end, int recurrences) {
+
+        LocalTime existingStartTime = this.start.toLocalTime();
+        LocalTime existingEndTime = this.end.toLocalTime();
+        DayOfWeek existingDay = this.start.getDayOfWeek();
+
+        LocalTime newStartTime = start.toLocalTime();
+        LocalTime newEndTime = end.toLocalTime();
+        DayOfWeek newDay = start.getDayOfWeek();
+
+        return (this.description.equals(description)
+                && existingStartTime == newStartTime
+                && existingEndTime == newEndTime
+                && existingDay == newDay
+                && ((this.recurrences > 0 && recurrences > 0) || this.start.equals(start)));
```

Comment:
Since this whole part is similar to lesson, maybe it can be factored out to be a common method to reduce duplicated code.

Reply:
I was thinking the same, but I was worried it would reduce the OOP aspect. I wanted to make this method inherit from Task, but couldn't because Deadlines don't store both the start and end times, but only the by.